### ISSUE TITLE
Add some `saveMap` and `loadMap` functions

### ISSUE THREAD

**Issue by rubiop** [MeasureTransport/MParT#267]
Reusing map computations is in the current state quite difficult. Due to the way matlab bindings are made (and it might be the same thing with python or julia) you can't directly save a map object. Coding a `save` function directly in C++ that could be use for all bindings might be useful. My first idea is just to write a file (maybe with the `.mt` extension) with all informations of the map we want to save (e.g. map type (triangular, composed, Affine), multiIndexSet, coefficients, mapOptions etc.).

Then it might be useful to create a `load` constructor of maps that can takes as input a saved map file.

This is a very naive way to do it, I'm open to any better idea you might have.

**Comment by dannys4** [comment]
I think this is a good idea-- maybe just put up a PR for internal functions `viewToStream` and `viewFromStream` first (dealing with input/output streams)-- that should easily lead to a `save` and `load` for the coeffs and the multiindex set, where you create a file stream, enter in the dimension and the number of coefficients, put the multiindex set into it, then put the coefficients into it (similar for loading). Implementing a `save` function for a whole map might be a bit more daunting, and (if I'm not mistaken) most of the information on the map is encoded in those two objects. This should help as a stopgap so that we have minimum working functionality now, and we can improve on it later.

**Comment by mparno** [comment]
This type of problem is where object serialization tools are key.  There are all sorts of subtleties to storing complicated objects like our maps (duplicating shared_ptrs, etc...) that I don't think we really want to mess with.   I would strongly advocate for using [Cereal](https://uscilab.github.io/cereal/) here.  It's super slick and will allow us to add a few functions to each class that will enable us to easily save/load objects of arbitrary complexity.    

Cereal is also what we used to write [ParCer](https://bitbucket.org/mituq/parcer/src/master/), which allows you to use MPI to send/recv any cereal-ized object between compute nodes.

**Comment by dannys4** [comment]
tl;dr-- this will be easier for stuff. Implement this "archive" function for each class, should make things easy to save+load. Make `MPART_BUILD_ARCHIVING` to make `cereal` mandatory.
